tests/test.main.js

```diff
@@ -0,0 +1,33 @@
+import { getConfig } from 'cli';
+import { createInstance, isRunFromCLI } from 'main';
+
+
+describe('Main module tests', function() {
+
+  it('should not use yargs when not called from CLI', () => {
+    assert.doesNotThrow(() => {
+      createInstance({
+        // Passing `null` as a second argument means we don't supply
+        // a valid `argv` module, so this will fail if the code tries to
+        // call `argv`.
+        config: getConfig(false, null).argv,
```

Comment:
it's my personal preference but I would much rather see `getConfig({useCLI=false, _argv: null})`. You have to type some extra characters but I think it makes things easier to read and easier to expand upon in the future.